src/app/common/hooks/use-register-children.ts

```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+
+export function useRegisterChildren<T extends string>() {
+  const [children, setChildren] = useState<Record<T, number>>({} as Record<T, number>);
+
+  function registerChild(child: T) {
+    setChildren(children => ({ ...children, [child]: (children[child] || 0) + 1 }));
+  }
```

Comment:
`issueBirthCertificate(childId) {`

Reply:
Lots of children in this PR. I started using this in the approver but removed, however kept the refactor for `Input` component